pkg/apis/apps/types.go

```diff
@@ -101,3 +102,38 @@ type StatefulSetList struct {
 	metav1.ListMeta
 	Items []StatefulSet
 }
+
+// MaxControllerRevisionSize is the maximum size of an ControllerRevision Object.
+const MaxControllerRevisionSize = 1 * 1024 * 1024
```

Comment:
I would not expose this constant here - it's a detail of storage in general, and not something clients should be referencing (server should reject, not client, so we can relax in the future).  Instead we typically define this in validation and only impose it there (see annotations).